seed/energy/meter_data_processor/kairos_insert.py

```diff
@@ -70,15 +73,26 @@ def batch_insert_kairosdb(meta_data, ts_data):
         insert_data['tags'] = meta
 
         wrap.append(insert_data)
+        if len(wrap) == KairosDB_Batch_Insert_Size:
+            json_insert_data = json.dumps(wrap)
+            r = requests.post(db_data['insert_url'], data=json_insert_data)
+            if r.status_code != 204:
+                _log.error('Insert Into KairosDB Error ' + str(r.status_code))
+                _log.info('Error Message: ' + r.text)
+                return False
+
+            counter += KairosDB_Batch_Insert_Size
+            _log.info('KairosDB_Inserted, ' + str(counter) + '/' + str(total_len))
+            wrap = []
+
+    if len(wrap) > 0:
+        json_insert_data = json.dumps(wrap)
+        r = requests.post(db_data['insert_url'], data=json_insert_data)
```

Comment:
I will follow the post and dry this up.

On Tue, Mar 29, 2016 at 12:55 PM, Nicholas Long <notifications@github.com>
wrote:

> In seed/energy/meter_data_processor/kairos_insert.py
> <https://github.com/SEED-platform/seed/pull/863#discussion_r57758560>:
>
> > @@ -70,15 +73,26 @@ def batch_insert_kairosdb(meta_data, ts_data):
> >          insert_data['tags'] = meta
> >
> >          wrap.append(insert_data)
> > +        if len(wrap) == KairosDB_Batch_Insert_Size:
> > +            json_insert_data = json.dumps(wrap)
> > +            r = requests.post(db_data['insert_url'], data=json_insert_data)
> > +            if r.status_code != 204:
> > +                _log.error('Insert Into KairosDB Error ' + str(r.status_code))
> > +                _log.info('Error Message: ' + r.text)
> > +                return False
> > +
> > +            counter += KairosDB_Batch_Insert_Size
> > +            _log.info('KairosDB_Inserted, ' + str(counter) + '/' + str(total_len))
> > +            wrap = []
> > +
> > +    if len(wrap) > 0:
> > +        json_insert_data = json.dumps(wrap)
> > +        r = requests.post(db_data['insert_url'], data=json_insert_data)
>
> can we DRY this up a bit? Either make the post a method (and check the
> response code) or write a chunk method. (
> http://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks-in-python
> )
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/SEED-platform/seed/pull/863/files/07fcd5c43cfd1d50ca513fd8bd38d7410b999749#r57758560>
>